config/test.exs

```diff
@@ -1,3 +1,5 @@
 use Mix.Config
 
+config :kafka_ex, snappy_module: :snappyer
```

Comment:
Again, please don't override this unless we plan to make snappyer the default.  You can (and we should) test snappyer by wrapping a test where we change the config.

Reply:
This is forced to use in the testenvironment and the ci environment.

Reply:
This changes the config for *all* the tests, meaning we're not protected against regressions when using the old snappy lib.

This is problematic, because it means that the test suite wouldn't cover *at all* the default config we're shipping (i.e. the old snappy lib).

I suggest modifying the tests that cover compression so that they run twice, once with the default config and one with the new snappy module.

You can change the Application config for a given test (and restore it afterwards) by using `Application.put_env`.

Reply:
do we even need to test snappyer? is this a case where we need to define a compression behavior(interface in other languages)  such that this is swappable in the future without any modification to kafka_ex?

Reply:
Not test snappyer

Reply:
> do we even need to test snappyer?

If you're talking about the lib itself, we don't. We do need to test the it integrates correctly, though right ? What I mean is: we do need to test that *our* code, when calling snappyer, works as expected.

Also, to re-iterate my earlier point: 
- most of our tests should use as much as possible the default config for the tests (unless we really need to), do be consistent with what we ship in a release.
- whenever there are several configs, we should try to have at least one test checking that the changed config behaves correctly.

>  is this a case where we need to define a compression behavior [...] ?

Interesting idea ! I don't know much about compression in Kafka, so I don't know how useful that would be.

Reply:
Addendum: the reason why I would recommend having one test with the new snappyer library is that, if we wish to make it the default for KafkaEx 1.0, we must have at least *some* experience with it. Including a test in the CI for snappyer allows us to get a bit of that.

Reply:
Sorry just seeing this, maybe I wasn't clear, my hope is that we don't have to always have people open PRs like this in the future, we provide them with a interface they need to implement and let them go implement the interface for any new library they want to use for compression, I may be too far away from things and my suggestions might just be flat out wrong, but it seems it's just easier to empower people to change the compression library by implementing an interface rather than always opening PR to kafka_ex every time there is a new shiny/better compression library/support in the elixir ecosystem or kafka.